client/core/trade.go

```diff
@@ -1873,7 +1851,7 @@ func (t *trackedTrade) refundMatches(matches []*matchTracker) (uint64, error) {
 			refundedQty += calc.BaseToQuote(match.Match.Rate, match.Match.Quantity)
 		}
 		proof.RefundCoin = []byte(refundCoin)
-		proof.SelfRevoked = true // Set match as revoked.
+		t.revokeMatch(match, false) // self-revoke this match
 		err = t.db.UpdateMatch(&match.MetaMatch)
```

Comment:
This is where I figure the match is already revoked by the server.  Maybe there should be a condition to check first if it's already revoked.